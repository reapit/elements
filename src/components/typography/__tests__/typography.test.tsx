import { render } from '@testing-library/react'
import {
  Title,
  Subtitle,
  BodyText,
  SmallText,
  Text2XS,
  TextXS,
  Text2XL,
  Text3XL,
  TextBase,
  TextL,
  TextSM,
  TextXL,
  TaggedTypography,
  TypeographyTag,
} from '..'

describe('Typography Base', () => {
  it('Text3XL should match a snapshot', () => {
    const wrapper = render(<Text3XL>I am a Text3XL</Text3XL>)
    expect(wrapper).toMatchSnapshot()
  })

  it('Text3XL should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <Text3XL
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am a Text3XL
      </Text3XL>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Text2XL should match a snapshot', () => {
    const wrapper = render(<Text2XL>I am a Text2XL</Text2XL>)
    expect(wrapper).toMatchSnapshot()
  })

  it('Text2XL should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <Text2XL
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am a Text2XL
      </Text2XL>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('TextXL should match a snapshot', () => {
    const wrapper = render(<TextXL>I am a TextXL</TextXL>)
    expect(wrapper).toMatchSnapshot()
  })

  it('TextXL should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <TextXL
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am a TextXL
      </TextXL>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('TextL should match a snapshot', () => {
    const wrapper = render(<TextL>I am a TextL</TextL>)
    expect(wrapper).toMatchSnapshot()
  })

  it('TextL should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <TextL
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am a TextL
      </TextL>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('TextBase should match a snapshot', () => {
    const wrapper = render(<TextBase>I am a TextBase</TextBase>)
    expect(wrapper).toMatchSnapshot()
  })

  it('TextBase should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <TextBase
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am a TextBase
      </TextBase>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('TextSM should match a snapshot', () => {
    const wrapper = render(<TextSM>I am a TextSM</TextSM>)
    expect(wrapper).toMatchSnapshot()
  })

  it('TextSM should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <TextSM
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am a TextSM
      </TextSM>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('TextXS should match a snapshot', () => {
    const wrapper = render(<TextXS>I am a TextXS</TextXS>)
    expect(wrapper).toMatchSnapshot()
  })

  it('TextXS should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <TextXS
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am a TextXS
      </TextXS>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Text2XS should match a snapshot', () => {
    const wrapper = render(<Text2XS>I am a Text2XS</Text2XS>)
    expect(wrapper).toMatchSnapshot()
  })

  it('Text2XS should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <Text2XS
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am a Text2XS
      </Text2XS>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Typography Helpers', () => {
  it('Title should match a snapshot', () => {
    const wrapper = render(<Title>I am a title</Title>)
    expect(wrapper).toMatchSnapshot()
  })

  it('SubTitle should match a snapshot', () => {
    const wrapper = render(<Subtitle>I am a subtitle</Subtitle>)
    expect(wrapper).toMatchSnapshot()
  })

  it('BodyText should match a snapshot', () => {
    const wrapper = render(<BodyText>I am body text</BodyText>)
    expect(wrapper).toMatchSnapshot()
  })

  it('SmallText should match a snapshot', () => {
    const wrapper = render(<SmallText>I am small text</SmallText>)
    expect(wrapper).toMatchSnapshot()
  })

  it('SmallText should match a snapshot', () => {
    const wrapper = render(<SmallText>I am small text</SmallText>)
    expect(wrapper).toMatchSnapshot()
  })

  it('Title should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <Title
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am a title
      </Title>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('SubTitle should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <Subtitle
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am a subtitle
      </Subtitle>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('BodyText should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <BodyText
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am body text
      </BodyText>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('SmallText should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <SmallText
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am small text
      </SmallText>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('SmallText should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <SmallText
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        I am small text
      </SmallText>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('SmallText should match a snapshot with all modifiers', () => {
    const wrapper = render(
      <SmallText
        hasGreyText
        hasRegularText
        hasBoldText
        hasMediumText
        hasItalicText
        hasNoMargin
        hasMargin
        hasDisabledText
        hasCenteredText
        hasSectionMargin
        hasCapitalisedText
        hasUpperCasedText
        intent="primary"
      >
        i am some text
      </SmallText>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  describe('TaggedTypography', () => {
    ;[...Array.from({ length: 5 }, (_, i) => i + 1).map((index) => `h${index}`), 'p', 'small', 'span', 'div'].forEach(
      (tag) => {
        it(`${tag}`, () => {
          const component = render(<TaggedTypography tag={tag as TypeographyTag}></TaggedTypography>)
          expect(component.baseElement.querySelector(tag)?.tagName.toString().toLowerCase()).toBe(tag)
        })
      },
    )

    describe('Typography Components', () => {
      ;[...Array.from({ length: 5 }, (_, i) => i + 1).map((index) => `h${index}`), 'p', 'small', 'span', 'div'].forEach(
        (tag) => {
          Object.entries({
            text3xl: <Text3XL tag={tag as TypeographyTag}></Text3XL>,
            text2xl: <Text2XL tag={tag as TypeographyTag}></Text2XL>,
            textxl: <TextXL tag={tag as TypeographyTag}></TextXL>,
            textl: <TextL tag={tag as TypeographyTag}></TextL>,
            textBase: <TextBase tag={tag as TypeographyTag}></TextBase>,
            textSm: <TextSM tag={tag as TypeographyTag}></TextSM>,
            textXs: <TextXS tag={tag as TypeographyTag}></TextXS>,
            text2Xs: <Text2XS tag={tag as TypeographyTag}></Text2XS>,
          }).forEach(([name, element]) => {
            it(`Element ${name} can return ${tag}`, () => {
              const component = render(element)
              expect(component.baseElement.querySelector(tag)?.tagName.toString().toLowerCase()).toBe(tag)
            })
          })
        },
      )
    })
  })
})
