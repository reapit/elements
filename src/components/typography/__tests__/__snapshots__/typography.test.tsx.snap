// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Typography Base > Text2XL should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text2xl"
  >
    I am a Text2XL
  </div>
</DocumentFragment>
`;

exports[`Typography Base > Text2XL should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text2xl el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am a Text2XL
  </div>
</DocumentFragment>
`;

exports[`Typography Base > Text2XS should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text2xs"
  >
    I am a Text2XS
  </div>
</DocumentFragment>
`;

exports[`Typography Base > Text2XS should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text2xs el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am a Text2XS
  </div>
</DocumentFragment>
`;

exports[`Typography Base > Text3XL should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text3xl"
  >
    I am a Text3XL
  </div>
</DocumentFragment>
`;

exports[`Typography Base > Text3XL should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text3xl el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am a Text3XL
  </div>
</DocumentFragment>
`;

exports[`Typography Base > TextBase should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="el-text-base"
  >
    I am a TextBase
  </div>
</DocumentFragment>
`;

exports[`Typography Base > TextBase should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am a TextBase
  </div>
</DocumentFragment>
`;

exports[`Typography Base > TextL should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text-l"
  >
    I am a TextL
  </div>
</DocumentFragment>
`;

exports[`Typography Base > TextL should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text-l el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am a TextL
  </div>
</DocumentFragment>
`;

exports[`Typography Base > TextSM should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text-sm"
  >
    I am a TextSM
  </div>
</DocumentFragment>
`;

exports[`Typography Base > TextSM should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text-sm el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am a TextSM
  </div>
</DocumentFragment>
`;

exports[`Typography Base > TextXL should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text-xl"
  >
    I am a TextXL
  </div>
</DocumentFragment>
`;

exports[`Typography Base > TextXL should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text-xl el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am a TextXL
  </div>
</DocumentFragment>
`;

exports[`Typography Base > TextXS should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text-xs"
  >
    I am a TextXS
  </div>
</DocumentFragment>
`;

exports[`Typography Base > TextXS should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <div
    class="el-text-base el-text-xs el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am a TextXS
  </div>
</DocumentFragment>
`;

exports[`Typography Helpers > BodyText should match a snapshot 1`] = `
<DocumentFragment>
  <p
    class="el-text-base el-body-text"
  >
    I am body text
  </p>
</DocumentFragment>
`;

exports[`Typography Helpers > BodyText should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <p
    class="el-text-base el-body-text el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am body text
  </p>
</DocumentFragment>
`;

exports[`Typography Helpers > SmallText should match a snapshot 1`] = `
<DocumentFragment>
  <small
    class="el-text-base el-small-text"
  >
    I am small text
  </small>
</DocumentFragment>
`;

exports[`Typography Helpers > SmallText should match a snapshot 2`] = `
<DocumentFragment>
  <small
    class="el-text-base el-small-text"
  >
    I am small text
  </small>
</DocumentFragment>
`;

exports[`Typography Helpers > SmallText should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <small
    class="el-text-base el-small-text el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am small text
  </small>
</DocumentFragment>
`;

exports[`Typography Helpers > SmallText should match a snapshot with all modifiers 2`] = `
<DocumentFragment>
  <small
    class="el-text-base el-small-text el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am small text
  </small>
</DocumentFragment>
`;

exports[`Typography Helpers > SmallText should match a snapshot with all modifiers 3`] = `
<DocumentFragment>
  <small
    class="el-text-base el-small-text el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    i am some text
  </small>
</DocumentFragment>
`;

exports[`Typography Helpers > SubTitle should match a snapshot 1`] = `
<DocumentFragment>
  <h2
    class="el-text-base el-subtitle"
  >
    I am a subtitle
  </h2>
</DocumentFragment>
`;

exports[`Typography Helpers > SubTitle should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <h2
    class="el-text-base el-subtitle el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am a subtitle
  </h2>
</DocumentFragment>
`;

exports[`Typography Helpers > Title should match a snapshot 1`] = `
<DocumentFragment>
  <h1
    class="el-text-base el-title"
  >
    I am a title
  </h1>
</DocumentFragment>
`;

exports[`Typography Helpers > Title should match a snapshot with all modifiers 1`] = `
<DocumentFragment>
  <h1
    class="el-text-base el-title el-has-grey-text el-has-regular-text el-has-bold-text el-has-italic-text el-has-medium-text el-has-margin el-has-no-margin el-has-section-margin el-has-centered-text el-has-disabled-text el-has-capitalised-text el-has-upper-cased-text el-intent-primary"
  >
    I am a title
  </h1>
</DocumentFragment>
`;
